File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map=new HashMap<>();
        int ans[]=new int[2];
        for(int i=0;i<nums.length;i++){
            int temp=target-nums[i];
            if(map.containsKey(temp)){
                ans[0]=map.get(temp);
                ans[1]=i;
                break;
            }
            else{
                map.put(nums[i],i);
            }
        }
        
        return ans;
    }
}

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    map={}
    ans=[]
   
    for(let i=0;i<nums.length;i++){
        if(target-nums[i] in map){
            ans.push(map[target-nums[i]])
            ans.push(i)
       //     return [map[target-nums[i]],i];
            break
        }
        else{
      
            map[nums[i]]=i
        }
    }
    return ans
};